From fb9a525871a304a186a849ac6f77f07a589952c5 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:21 -0400
Subject: [ethernet] be2net: avoid SRIOV config for BE2 chip

Message-id: <1410192881-30702-59-git-send-email-ivecera@redhat.com>
Patchwork-id: 92454
O-Subject: [RHEL7.1 PATCH 58/78] be2net: avoid SRIOV config for BE2 chip
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 962bcb750b47ef0c8d28cc217ec22b4e44413565
Author: Vasundhara Volam <vasundhara.volam@emulex.com>
Date:   Thu Jul 17 16:20:30 2014 +0530

    be2net: avoid SRIOV config for BE2 chip

    As SRIOV is not supported on BE2 chip, avoid calling be_get/set_sriov_config()
    for BE2 chip.

    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index fcb7dc1..ca1e6b9 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3430,7 +3430,9 @@ static int be_get_config(struct be_adapter *adapter)
   if (!status)
    dev_info(&adapter->pdev->dev,
      "Using profile 0x%x\n", profile_id);
+ }
 
+ if (!BE2_chip(adapter) && be_physfn(adapter)) {
   status = be_get_sriov_config(adapter);
   if (status)
    return status;
-- 
1.7.1